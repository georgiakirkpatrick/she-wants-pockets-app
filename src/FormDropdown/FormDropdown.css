* {
    box-sizing: border-box;
    padding: 0;
    margin: 0;
}

.FormDropdown {
    /* padding: 10px 0; */
    /* text-align: left; */
    width: 100%;
    margin-top: 10px;
}

.FormDropdown > select {
    display: block;
    width: 100%;
    margin: 2px 0px 10px;
    padding: 10px;
    font: inherit;
}    