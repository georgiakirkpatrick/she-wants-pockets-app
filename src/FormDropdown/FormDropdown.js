import React from 'react'
import './FormDropdown.css'

const FormDropdown = props => {
    
    
    return (
        <div key={props.id} className='FormDropdown'>
            <label htmlFor={props.id}>{props.prompt}</label>

            <select name={props.name} onChange={props.change} value={props.currentValue} >
                {props.options.map((opt) => {
                    return <option key={opt.id} id={opt.id} value={opt.value}>{opt.text}</option>
                })}
            </select>
        </div>
    )
}

export default FormDropdown