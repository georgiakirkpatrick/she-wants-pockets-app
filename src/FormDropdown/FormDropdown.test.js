import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import FormDropdown from './FormDropdown';

describe.only('FormDropdown', () => {
        it ('renders without crashing', () => {
            const div = document.createElement('div');
            ReactDOM.render(<FormDropdown />, div);
            ReactDOM.unmountComponentAtNode(div);
        });
    
        it ('render the UI as expected', () => {
            const tree = renderer
                .create(<FormDropdown />)
                .toJSON();
            expect(tree).toMatchSnapshot()
        });
    });