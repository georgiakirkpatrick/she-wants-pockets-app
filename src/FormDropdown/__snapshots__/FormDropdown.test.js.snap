// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormDropdown render the UI as expected 1`] = `
<div>
  <label />
</div>
`;
